sourcegraph/sourcegraph
# Code Insights" Migrate live preview handler to hooks

Part of https://github.com/sourcegraph/sourcegraph/issues/31085

## Background 

In this PR we migrate live preview handlers (search-based, capture group, compute based insights live previews) to hooks, previously they were built as a part of code insights backend service. For more context on why we should remove backend service class read in [the tracking issue](https://github.com/sourcegraph/sourcegraph/issues/41145) but TL;DR live preview hook should be easy to reuse which is important if we would want to integrate insights with other UI. 

## Test plan
- Manually test search based, capture group and compute based insights creation UI. 
- Check that live preview card doesn't have any regressions in lights of data fetching states (loading, error, actually data, intact state when we have blur chart and initial message)
- Check dynamic insight preview card on the getting started insight page.



## App preview:

- [Web](https://sg-web-vk-migrate-live-preview-to-hooks.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
